src/include/sof/lib/dma.h

```diff
@@ -564,19 +570,56 @@ static inline bool dma_is_scheduling_source(struct dma_chan_data *channel)
 	return ret;
 }
 
-static inline void dma_sg_init(struct dma_sg_elem_array *ea)
+static inline void dma_sg_init(struct dma_sg_array *sg_array)
+{
+	sg_array->count = 0;
+	sg_array->elems = NULL;
+}
+
+static inline void dma_sg_elems_init(struct dma_sg_elem_array *elem_array)
 {
-	ea->count = 0;
-	ea->elems = NULL;
+	elem_array->count = 0;
+	elem_array->elems = NULL;
 }
 
+static inline void dma_sg_elem_trace(struct dma_sg_elem *sg_elem)
+{
+	trace_event(TRACE_CLASS_DMA, "dma_sg_elem_trace() src %X dest %X size %d",
+		    sg_elem->src, sg_elem->dest, sg_elem->size);
+}
+
+static inline void dma_sg_elems_trace(struct dma_sg_elem_array *elem_array)
+{
+	int i;
+
+	trace_event(TRACE_CLASS_DMA, "dma_sg_elems_trace() count %d",
+		    elem_array->count);
+
+	for (i = 0; i < elem_array->count; i++)
+		dma_sg_elem_trace(&elem_array->elems[i]);
+}
+
+static inline void dma_sg_trace(struct dma_sg_array *sg_array)
+{
+	int i;
+
+	trace_event(TRACE_CLASS_DMA, "dma_sg_trace() count %d", sg_array->count);
+
+	for (i = 0; i < sg_array->count; i++)
+		dma_sg_elems_trace(&sg_array->elems[i]);
+}
```

Comment:
Debug function? I would like an argument over how useful this code would be in deployed builds.

Reply:
Will be removed